python/ray/serialization.py

```diff
@@ -201,6 +206,29 @@ def _deserialize_object(self, data, metadata, object_ref):
                     0] == ray_constants.OBJECT_METADATA_TYPE_ACTOR_HANDLE:
                 obj = self._deserialize_msgpack_data(data, metadata_fields)
                 return _actor_handle_deserializer(obj)
+            elif metadata_fields[
```

Comment:
I don't think we need a new metadata type, it can be embedded within msgpack like pickle5 data is.

Reply:
@iycheng can chime in more. The reason here is to avoid double serialization from protobuf -> bytes -> msgpack wrapped bytes.

Reply:
Yes, this can avoid double serialization and also is useful to support zero-copy semantic in the future with flatbuffer

Reply:
@ericl is this a blocker issue for you? I can push it into msgpack as well (in fact that's the previous implementation 😅)

Reply:
The pickle5 implementation actually avoids double serialization, by storing the pickle5 serialized data separately to enable zero-copy.

I'd like us to settle on a common implementation for these types of extensions, as we are seeing ArrowTable / other serialization extensions being proposed: https://github.com/ray-project/ray/pull/20242

@jovany-wang is leading the x-lang serialization improvements for 2.0; I've started a thread in Ray slack

Reply:
Sounds good. I have updated the PR to have the data wrapped by msgpack instead. We can't do exactly separately serialization like the way pickle5 did it because currently messagepack serializer defaults to pickle5 and there is no way to specify a mapping of type=>serializer for the msgpack extension types. This part needs more discussion.